packages/extension-core/src/domains/analytics/store.ts

```diff
@@ -0,0 +1,142 @@
+import * as Sentry from "@sentry/browser"
+import { DEBUG, log } from "extension-shared"
+import { v4 } from "uuid"
+
+import { StorageProvider } from "../../libs/Store"
+import { remoteConfigStore } from "../app/store.remoteConfig"
+import { settingsStore } from "../app/store.settings"
+import { PostHogCaptureProperties } from "./types"
+
+export type AnalyticsEvent = {
+  eventName: string
+  properties?: PostHogCaptureProperties
+  timestamp: number
+}
+
+type AnalyticsData = { data: AnalyticsEvent[]; distinctId: string }
+
+const TALISMAN_PROPERTIES = {
+  appVersion: process.env.VERSION,
+  appBuild: process.env.BUILD,
+  testBuild: DEBUG || ["dev", "qa", "ci"].includes(process.env.BUILD as string),
+}
+
+class AnalyticsStore extends StorageProvider<AnalyticsData> {
+  // the isReady promise prevents events being added to the queue while the analytics are being sent
+  isReady: Promise<boolean> = new Promise(() => true)
+
```

Comment:
This never resolves, so i couldnt get my tests to work

so i tested with this instead: 
```ts
isReady: Promise<boolean> = new Promise((resolve) => { resolve(true) })
```

Reply:
after i did i had this error once: 
![image](https://github.com/user-attachments/assets/4eaf7fac-acfa-42f7-bab7-52293f2ec5d6)
![image](https://github.com/user-attachments/assets/e815a833-f9d2-4e6b-a87e-b008c1e05a9d)